hazelcast/src/main/java/com/hazelcast/internal/probing/Probing.java

```diff
@@ -0,0 +1,205 @@
+package com.hazelcast.internal.probing;
+
+import static com.hazelcast.internal.probing.CharSequenceUtils.startsWith;
+import static com.hazelcast.internal.probing.ProbeRegistry.ProbeSource.TAG_INSTANCE;
+import static com.hazelcast.internal.probing.ProbeRegistry.ProbeSource.TAG_TARGET;
+import static com.hazelcast.internal.probing.ProbeRegistry.ProbeSource.TAG_TYPE;
+import static java.lang.Math.round;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.concurrent.Semaphore;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import com.hazelcast.internal.probing.CharSequenceUtils.Lines;
+import com.hazelcast.internal.probing.ProbeRegistry.ProbeSource;
+import com.hazelcast.internal.probing.ProbingCycle.Tags;
+import com.hazelcast.internal.util.counters.Counter;
+import com.hazelcast.monitor.LocalIndexStats;
+import com.hazelcast.monitor.NearCacheStats;
+import com.hazelcast.monitor.impl.LocalDistributedObjectStats;
+import com.hazelcast.monitor.impl.LocalMapStatsImpl;
+
+/**
+ * Provides utilities around probing.
+ * 
+ * This includes general conversion functionality like {@link #toLong(boolean)}
+ * as well as common metrics on core types of objects that cannot be annotated.
+ */
+public final class Probing {
```

Comment:
Please call this class ProbeUtils. 

To indicate it is a utility class.

Reply:
done.